lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -230,8 +245,34 @@ void AtenSizeOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
       listElements.push_back(rewriter.create<::mlir::ConstantOp>(
           op->getLoc(), rewriter.getI64IntegerAttr(size)));
     }
-    rewriter.replaceOpWithNewOp<Basicpy::BuildListOp>(
-        op, Basicpy::ListType::get(rewriter.getContext()), listElements);
+    rewriter.replaceOpWithNewOp<Torch::PrimListConstructOp>(
+        op, Torch::ListType::get(rewriter.getI64Type()), listElements);
+    return success();
+  });
+}
+
+//===----------------------------------------------------------------------===//
+// Aten__Getitem__TOp
+//===----------------------------------------------------------------------===//
+
+void Aten__Getitem__TOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                     MLIRContext *context) {
+  patterns.add(+[](Aten__Getitem__TOp op, PatternRewriter &rewriter) {
+    auto torchList = op.getOperand(0);
+    if(!torchList.hasOneUse())
+      return failure();
+
+    auto listConstruct = torchList.getDefiningOp<Torch::PrimListConstructOp>();
+    if (!listConstruct)
+      return failure();
+
+    auto constantIndexOp = op.getOperand(1).getDefiningOp<::mlir::ConstantOp>();
+    if (!constantIndexOp)
+      return failure();
+
+    auto attr = constantIndexOp->getAttrOfType<::mlir::IntegerAttr>("value");
```

Comment:
nit: You should be able to use `m_ConstantInt` for matching the constant op (see [example](https://github.com/llvm/mlir-npcomp/blob/b7b7fd4959e1834274206d4c2666ebd87165bc85/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L280))